model/crud/crud.go

```diff
@@ -0,0 +1,29 @@
+// Package crud provides a crud interface
+package crud
+
+type CRUD interface {
+	// Read values
+	Read(...ReadOption) ([]Record, error)
```

Comment:
[]*Record?

```suggestion
	Read(...ReadOption) ([]*Record, error)
```

Reply:
Record is an interface, should it be a struct? The idea is that it provides a method to scan in the value. Otherwise Record value would be an interface that's not actually the users struct type?

Reply:
I mean we could do it like https://golang.org/pkg/database/sql/#Row.Scan

Reply:
Ah, that makes sense